mautic/mautic
# Send example as contact

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
These feature extend send example email and improve email testing.  Now it's possible search contact and send example email with tokens related to that contact. All selected values are saved to cookie and remember it on each action.

Thanks for http://ballistix.com/ for support this PR

![image](https://user-images.githubusercontent.com/462477/40107529-c7e0632c-58f8-11e8-8004-f9b9869c77f8.png)

#### Steps to test this PR:
1.  Create email with tokens {contactfield=firstname} and {contactfield=lastname}
2.  Try Send example, try find related contact for test purpose and try send email yourself.
3. See if  you received example email with selected contacts tokens.

Before:

![image](https://user-images.githubusercontent.com/462477/40107765-69d23b42-58f9-11e8-8f9c-713ec5cd0676.png)

After PR 

![image](https://user-images.githubusercontent.com/462477/40107771-70e7fdf4-58f9-11e8-93bd-1ccf12d2149b.png)
